common/lib/common-utils/src/disposal.ts

```diff
@@ -3,7 +3,33 @@
  * Licensed under the MIT License.
  */
 
-import { IDisposable } from "@fluidframework/common-definitions";
+/**
+ * Base interface for objects that require lifetime management via explicit disposal.
+ *
+ * @deprecated Only exists because of {@link doIfNotDisposed} which is also deprecated and will be removed at some
+ * point. Don't use this.
+ *
+ * @internal
+ *
+ * @privateremarks
+ * This was copied here from common-definitions when we removed that package.
+ * It only exists because of doIfNotDisposed.
+ * When that function (unused in the repo) is removed, this should go away with it.
```

Comment:
I added notes like this one to all the things that were copied from common-definitions.